src/fec_base.h

```diff
@@ -223,7 +248,16 @@ class FecCode {
         vec::Vector<T>* vx,
         int* vx_zero = 0);
 
-    virtual void decode_Lagrange(
+    virtual void decode_lagrange(
+        vec::Vector<T>* output,
+        const std::vector<Properties>& props,
+        off_t offset,
+        vec::Vector<T>* fragments_ids,
+        vec::Vector<T>* words,
+        vec::Vector<T>* vx,
+        int vx_zero = 0);
```

Comment:
Do you need the default value here?

It seems that default value can have surprising behavior when used with virtual function and it's not recommanded to mix both. See https://google.github.io/styleguide/cppguide.html#Default_Arguments

If possible, we should avoid to mix virtual and default.

(same apply for `decode_prepare`).